utils/server.ts

```diff
@@ -15,18 +15,60 @@ const getAuthorizationHeader = async () => {
   }
 };
 
+const objectToFormData = (
+  obj: Record<string, any>,
+  formData: FormData = new FormData(),
+  parentKey: string = ''
+): FormData =>
+  Object.entries(obj).reduce((fd: FormData, [key, value]: [string, any]) => {
+    const propName: string = parentKey ? `${parentKey}[${key}]` : key;
+
+    if (typeof value === 'object' && !(value instanceof File)) {
+      return objectToFormData(value, fd, propName);
+    } else if (Array.isArray(value)) {
+      value.forEach((item: any, index: number) =>
+        objectToFormData(item, fd, `${propName}[${index}]`)
+      );
+    } else {
+      if (value !== undefined) fd.append(propName, value);
+    }
+
+    return fd;
+  }, formData);
+
+const buildRequestBody = (
+  body: Record<string, any>,
+  sendAsFormData: boolean
+) => {
+  if (sendAsFormData) {
+    return objectToFormData(body);
+  }
+  return JSON.stringify(body);
+};
+
 export const makeRequest = async ({
   path,
   method = 'GET',
   body,
+  headers,
+  sendAsFormData,
 }: {
   path: string;
   method: string;
   body?: Record<string, any>;
+  headers?: Record<string, string>;
+  sendAsFormData?: boolean;
 }) => {
   try {
     const requestHeaders: HeadersInit = new Headers();
-    requestHeaders.set('Content-Type', 'application/json');
+
+    for (const [key, value] of Object.entries(
+      headers || sendAsFormData ? {} : { 'Content-Type': 'application/json' }
+    )) {
+      requestHeaders.set(key, value);
+    }
```

Comment:
I would prefer using `forEach`, but this is fine as well (although the ternary operator expression placed within the `for` loop header is a bit confusing at a first glance).

```ts
const checkedHeaders = headers || sendAsFormData ? {} : { 'Content-Type': 'application/json' };
Object.keys(checkedHeaders).forEach((key: string) => requestHeaders.set(key, checkedHeaders[key]));
```